optimizely/config/datafileprojectconfig/config.go

```diff
@@ -82,11 +87,16 @@ func NewDatafileProjectConfig(jsonDatafile []byte) (*DatafileProjectConfig, erro
 		return nil, err
 	}
 
+	attributeMap, attributeKeyToIDMap := mappers.MapAttributes(datafile.Attributes)
 	experiments, experimentKeyMap := mappers.MapExperiments(datafile.Experiments)
 	rolloutMap := mappers.MapRollouts(datafile.Rollouts)
 	mergedAudiences := append(datafile.TypedAudiences, datafile.Audiences...)
 	config := &DatafileProjectConfig{
 		audienceMap:          mappers.MapAudiences(mergedAudiences),
+		attributeMap:         attributeMap,
+		attributeKeyToIDMap:  attributeKeyToIDMap,
+		clientName:           optimizely.ClientName,
```

Comment:
`clientName` isn't in the datafile - should it be part of the instance or `Optimizely` instead?

Reply:
Ah yeah good point. I'll move it out and maybe pass it to the event factory since that is where it is needed